lib/options-manager.js

```diff
@@ -296,7 +296,12 @@ const buildXOConfig = options => config => {
 	}
 
 	if (options.semicolon === false && !options.prettier) {
-		config.rules.semi = ['error', 'never'];
+		if (options.ts) {
```

Comment:
This would cause the same issue as with the `indent` rule that you reported in #445.

This code should be:

```
if (options.semicolon === false && !options.prettier) {
  config.rules.semi = ['error', 'never'];
  config.rules['semi-spacing'] = ['error', {
    before: false,
    after: true
  }];

  if (options.ts) {
    config.rules['@typescript-eslint/semi'] = ['error', 'never'];
  }
}
```

Reply:
The semi rule should be disabled for typescript linting:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/semi.md

Reply:
But you're right, the base 'semi' rule has to be overridden

Reply:
Actually… why?
The semi rule is already set 'off' [here](https://github.com/xojs/eslint-config-xo-typescript/blob/e69953688059fabf2df8c29b18b8b8f74fe9f0f3/index.js#L407).

This does seem to work when the branch is not merged with the current master.
When merging it to the master the 'semi' rule is failing but it should be overridden to be 'off'.
Not sure what the reason for it is.

I used `npm link` on this branch and linked it into a small testing project with an index.ts and index.js
```js
export function foo(): void {
	console.log('bar')
}
```

As soon as I merge master into this branch the 'semi' rule appears for the index.ts which shouldn't occur. There is something else still wrong.

Reply:
The problem is related to the order of `extends`. Currently `eslint-config-xo-typescript` is imported before `eslint-config-xo`.
I'll look at changing that.
I think ideally it should be: `eslint-config-xo`, then `eslint-config-xo-typescript` then users `extends`

Reply:
will be fixed with #453